contracts/utils/AccessControlExtended.sol

```diff
@@ -11,70 +11,88 @@ import "./AccessControl.sol";
  */
 contract AccessControlExtended is AccessControl {
     /**
-     * @dev Modifier that restricts access to addresses having roles with the given role name and chain slug.
-     * Throws an error if the caller does not have permit.
-     * @param roleName_ The name of the role.
-     * @param chainSlug_ The chain slug associated with the role.
+     * @dev Constructor that sets the owner of the contract.
      */
-    modifier onlyRoleWithChainSlug(
-        string memory roleName_,
-        uint256 chainSlug_
-    ) {
-        bytes32 role = keccak256(abi.encode(roleName_, chainSlug_));
-        if (!_hasRole(role, msg.sender)) revert NoPermit(role);
-        _;
-    }
+    constructor(address owner_) AccessControl(owner_) {}
 
     /**
-     * @dev Constructor that sets the owner of the contract.
+     * @dev Checks if an address has the role.
+     * @param roleName_ The name of the role.
+     * @param chainSlug_ The chain slug associated with the role.
+     * @param address_ The address to be granted the role.
      */
-    constructor(address owner_) AccessControl(owner_) {}
+    function _checkRoleWithSlug(
+        bytes32 roleName_,
+        uint256 chainSlug_,
+        address address_
+    ) internal virtual {
+        bytes32 roleHash = keccak256(abi.encode(roleName_, chainSlug_));
+        if (!_hasRole(roleHash, address_)) revert NoPermit(roleHash);
+    }
 
     /**
      * @dev Grants a role to an address based on the role name and chain slug.
      * @param roleName_ The name of the role.
      * @param chainSlug_ The chain slug associated with the role.
      * @param grantee_ The address to be granted the role.
      */
-    function grantRole(
-        string memory roleName_,
-        uint256 chainSlug_,
+    function grantRoleWithSlug(
+        bytes32 roleName_,
+        uint32 chainSlug_,
         address grantee_
     ) external virtual onlyOwner {
-        _grantRole(roleName_, chainSlug_, grantee_);
+        _grantRoleWithSlug(roleName_, chainSlug_, grantee_);
     }
 
     /**
      * @dev Grants multiple roles to multiple addresses in batch.
      * @param roleNames_ The names of the roles to grant.
+     * @param slugs_ The slugs for chain specific roles. For roles which are not chain-specific, we can use slug = 0
      * @param grantees_ The addresses to be granted the roles.
      */
     function grantBatchRole(
         bytes32[] calldata roleNames_,
+        uint32[] calldata slugs_,
         address[] calldata grantees_
     ) external virtual onlyOwner {
         require(roleNames_.length == grantees_.length);
-        for (uint256 index = 0; index < roleNames_.length; index++)
-            _grantRole(roleNames_[index], grantees_[index]);
+        for (uint256 index = 0; index < roleNames_.length; index++) {
```

Comment:
doesn't check length of `slugs_[]`  (same for all comparable functions)

Reply:
Done in https://github.com/SocketDotTech/socket-DL/pull/234